src/cogent3/app/composable.py

```diff
@@ -702,9 +697,14 @@ def _as_completed(self, dstore, parallel=False, par_kw=None, **kwargs) -> Genera
     return ui.series(to_do, count=len(mapped), **kwargs)
 
 
-def is_composable(obj):
-    """checks whether obj has been registered by the composable decorator"""
-    return __app_registry.get(get_object_provenance(obj), False)
+def is_app_composable(obj):
+    """checks whether obj has been decorated by define_app and it's app_type attribute is not NON_COMPOSABLE"""
+    return hasattr(obj, "app_type") and obj.app_type is not NON_COMPOSABLE
```

Comment:
make this function more robust to potential future changes in decorator behaviour, change line to
```python
return is_app(obj) and obj.app_type is not NON_COMPOSABLE
```